src/CabanaPD_HeatTransfer.hpp

```diff
@@ -0,0 +1,101 @@
+/****************************************************************************
+ * Copyright (c) 2022-2023 by Oak Ridge National Laboratory                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of CabanaPD. CabanaPD is distributed under a           *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef HEATTRANSFER_H
+#define HEATTRANSFER_H
+
+#include <CabanaPD_Timer.hpp>
+
+namespace CabanaPD
+{
+
+// Peridynamic heat transfer with forward-Euler time integration.
+// Inherits only because this is a similar neighbor-based kernel.
+template <class ExecutionSpace, class ModelType>
+class HeatTransfer : public Force<ExecutionSpace, BaseForceModel>
+{
+  protected:
+    using base_type = Force<ExecutionSpace, BaseForceModel>;
+    using base_type::_half_neigh;
+    using base_type::_timer;
+    ModelType _model;
+
+  public:
+    HeatTransfer()
+        : base_type( false ){};
+
+    HeatTransfer( const bool half_neigh, const ModelType model )
+        : base_type( half_neigh )
+        , _model( model )
+    {
+    }
+
+    template <class TemperatureType, class PosType, class ParticleType,
+              class NeighListType, class ParallelType>
+    void
+    computeHeatTransferFull( TemperatureType& temp, const PosType& x,
+                             const PosType& u, const ParticleType& particles,
+                             const NeighListType& neigh_list, const int n_local,
+                             ParallelType& neigh_op_tag, const double dt )
```

Comment:
It seems this is not the full analogue of  "ComputeForceFull" since we do two things at the same time here: (i) sum over the particles and (ii) perform a time step integration.

Do you think this may be confusing?

Reply:
Probably, but I don't think it's worth the effort of separating right now. What other integration strategy would you plan to use?

Reply:
We would need to separate based on the comment below.